ml-agents/mlagents/trainers/learn.py

```diff
@@ -388,6 +390,14 @@ def create_environment_factory(
     env_args: Optional[List[str]],
 ) -> Callable[[int, List[SideChannel]], BaseEnv]:
     if env_path is not None:
+        if not path.exists(env_path):
```

Comment:
We actually don't launch the environment using the exact path -- there is some slightly more involved logic.  Maybe you could refactor `UnityEnvironment` slightly to make sure we use the same path here when checking for existence as is used in `executable_launcher(...)`: https://github.com/Unity-Technologies/ml-agents/blob/master/ml-agents-envs/mlagents_envs/environment.py#L152

Reply:
Maybe I'm a bit confused. Isn't this the same `env_path` that is within the `env_factory` and passed as input to the `SubprocessEnvManager`?

Reply:
The argument to `--env` might not be an actual file. If I build the scene as "Project/Build/grid" and run 
> `mlagents-learn config/trainer_config.yaml --train --env=Project/Build/grid`

it works. But `Project/Build/grid` doesn't exist.
```
$ ls Project/Build/grid
ls: Project/Build/grid: No such file or directory
$ ls Project/Build/grid.app/
Contents
```